unified-runtime/source/adapters/level_zero/command_buffer.cpp

```diff
@@ -830,6 +873,7 @@ urCommandBufferReleaseExp(ur_exp_command_buffer_handle_t CommandBuffer) {
   if (!CommandBuffer->RefCount.decrementAndTest())
     return UR_RESULT_SUCCESS;
 
+  UR_CALL(waitForOngoingExecution(CommandBuffer));
```

Comment:
If we are using `waitForOngoingExecution()` on the waitEvent path, we are waiting on the Fence instead of the Event. Doesn't that mean that the `SignalCommandList` is potentially still running by the time we destroy the command buffer? 

I think that could cause issues with events being destroyed and then reset during SignalCommandList execution. Probably the safest thing would be to wait on both the fence and the event.